### PR TITLE
Fixed deprecated github actions that the solc-bin PR check

### PR DESCRIPTION
**Pull request by BraxWong**
Have changed from actions/checkout@v2 to actions/checkout@v3 which should eliminate the warning message. actions/setup-python and actions/upload-artifact are not found within the .github folder.

### CODE REVIEW

**Comment by r0qs** [comment]
Hi @BraxWong thanks for your contribution but I guess there is a misunderstanding here, the actions in questions are related with the [solc-bin ](https://github.com/ethereum/solc-bin/tree/gh-pages) repository, please see here: https://github.com/ethereum/solc-bin/blob/gh-pages/.github/workflows/nightly-emscripten.yml.

We don't keep track of issues at that repository, since it is mainly used to host the current and historical compiler builds. So all issues are reported in the solidity repository but the PR should be done in the solc-bin repository. Could you please close this one and add your changes there? Also, please add the following: `Fixes https://github.com/ethereum/solidity/issues/14059` to the PR description, so the issue can be automatically closed when we merge it ;)

See our [contributing guidelines](https://docs.soliditylang.org/en/latest/contributing.html) for more information.

File: .circleci/config.yml [hunk 1]
```diff
@@ -265,7 +265,7 @@ defaults:
   - base_archlinux_large: &base_archlinux_large
       docker:
         - image: archlinux:base
-      resource_class: large
+      resource_class: macos.m1.large.gen1
```
Comment (r0qs) [inline .circleci/config.yml:268]: This is a unrelated change. Please do it in a different PR. See my comment here: https://github.com/ethereum/solidity/pull/14159#issuecomment-1523711323

Please, read our contributing guides before open many PRs: https://github.com/ethereum/solidity/pull/14159#issuecomment-1523437136

**Review by r0qs — CHANGES_REQUESTED** [review]

**Comment by BraxWong** [comment]
Sorry about the extra commit. I am not too sure why this commit ended up here instead of giving me the option to create a new pull request. I will push the changes regarding the deprecated GitHub actions to the appropriate repository.

**Comment by r0qs** [comment]
Thanks! :)
